src/factory/cosmosCDCService.ts

```diff
@@ -9,45 +10,90 @@ import {
   getChangeFeedIteratorOptions,
   processChangeFeed,
 } from "../capturer/cosmos/cosmos";
+import { createContainer } from "../capturer/cosmos/utils";
 import { cosmosDBService } from "./cosmosDBService";
-import { ICDCService } from "./service";
+import { ICDCService, Item } from "./service";
+
+export const LEASE_CONTAINER_NAME = "cdc-data-lease";
 
 export const cosmosCDCService = {
   processChangeFeed:
     (
       client: CosmosClient,
       database: string,
       resource: string,
+      mqueueClient: KafkaProducerCompact<unknown>,
       leaseResource?: string,
       prefix?: string,
     ) =>
     (cosmosDBServiceClient: typeof cosmosDBService): TaskEither<Error, void> =>
       pipe(
-        E.Do,
-        E.bind("database", () =>
+        TE.Do,
+        // Connecting to the database
+        TE.bind("database", () =>
           cosmosDBServiceClient.getDatabase(client, database),
         ),
-        TE.fromEither,
+        // Checking that the container {resource} exists
         TE.bind("container", ({ database }) =>
           cosmosDBServiceClient.getResource(database, resource),
         ),
+        // Checking that the lease container {leaseResource} exists
+        // If not, a default lease container will be created in the next steps
         TE.bind("leaseContainer", ({ database }) =>
-          cosmosDBServiceClient.getResource(database, leaseResource),
+          pipe(
+            O.fromNullable(leaseResource),
+            O.fold(
+              () => TE.right<Error, O.Option<Container>>(O.none),
+              (lease) =>
+                pipe(
+                  cosmosDBServiceClient.getResource(database, lease),
+                  TE.map((lcontainer) => O.some(lcontainer)),
+                ),
+            ),
+          ),
         ),
+        // Checking from the lease container {leaseResource} that there is a continuationToken stored
         TE.bind("continuationToken", ({ leaseContainer }) =>
-          cosmosDBServiceClient.getItemByID(leaseContainer, prefix),
+          pipe(
+            leaseContainer,
+            O.fold(
+              () => TE.right<Error, O.Option<Item>>(O.none),
```

Comment:
do we need explicit typings here?

Reply:
O.Option needs at least 1 type argument

Reply:
What I mean is: we can safely remove `<Error, O.Option<Item>>` :)